2단계 요구사항 정리

mkkim90 opened the pull request:
안녕하세요 2단계 요구사항 정리 완료하여 리뷰 요청드립니다 🙇

mkkim90 committed:
feat : 상품요구사항 / 메뉴 요구사항 / 메뉴 그룹 요구 사항 / 주문 요구 사항 작성

mkkim90 committed:
feat : 주문 테이블 요구사항을 작성한다

mkkim90 committed:
refactor : 체크 박스 추가

mkkim90 committed:
refactor : 순번 제목 내용 구분을 통해 가독성 높이기

mkkim90 committed:
refactor : 체크 박스 추가

MaroKim commented on README.md:
```diff
@@ -8,7 +8,130 @@ docker compose -p kitchenpos up -d
+- [ ] 가격은 필수값이며, 0보다 커야한다. 
+- [ ] 수정하고자 하는 상품이 존재해야한다.
+- [ ] 수정한 가격이 해당 상품이 포함된 메뉴 가격에 반영되어야 한다. 
+  - [ ] 합산 가격이 일치하지 않으면 해당 상품을 비노출 처리한다.
```
합산 가격이 어떻게 일치 하지 않을까요? 
좀더 자세히 작성해주시면 좋을 것 같아요. 
글로 설명이 어려운 부분은 부등호로 표현하는 방법도 좋습니다. :)

MaroKim commented on README.md:
```diff
@@ -8,7 +8,130 @@ docker compose -p kitchenpos up -d
+- [ ] 메뉴 상품은 필수값이며, 최소 1개 이상이어야ㅍ한다.
+- [ ] 메뉴 상품은 반드시 상품 테이블에 존재해야 한다.
+- [ ] 메뉴 상품의 수량은 0보다 커야한다.
+- [ ] 메뉴 가격은 메뉴 상품들의 가격 합과 일치해야 한다.
```
이 가격의 조건은 다시 한번 봐주시길 부탁드리겠습니다 :)

MaroKim commented on README.md:
```diff
@@ -8,7 +8,130 @@ docker compose -p kitchenpos up -d
+- [ ] 수정하려는 메뉴가 존재해야 한다.
+- [ ] 수정된 가격이 메뉴 상품들의 가격 합과 일치해야 한다.
+
+### 메뉴 노출
+- [ ] 노출하려는 메뉴가 존재해야 한다.
+- [ ] 메뉴 가격이 메뉴 상품들의 가격 합과 일치해야 한다.
+
+### 메뉴 비노출
+- [ ] 비노출하려는 메뉴가 존재해야 한다.
+
```
메뉴를 보여주고 안보여주는 것을 '노출', '비노출'로 정의 해주셨네요 :) 👍

MaroKim commented on README.md:
```diff
@@ -8,7 +8,130 @@ docker compose -p kitchenpos up -d
+### 2. 메뉴 관리
+
+#### 메뉴 생성
+- [ ] 가격은 필수값이며, 0보다 커야한다. 
+- [ ] 생성하려는 하는 메뉴의 그룹이 존재해야한다.
+- [ ] 메뉴 상품은 필수값이며, 최소 1개 이상이어야ㅍ한다.
+- [ ] 메뉴 상품은 반드시 상품 테이블에 존재해야 한다.
+- [ ] 메뉴 상품의 수량은 0보다 커야한다.
```
요구사항 전반적으로  긍정문으로 통일해서 작성해 주신것 👍 
읽기가 편했습니다 :)

MaroKim commented on README.md:
```diff
@@ -8,7 +8,130 @@ docker compose -p kitchenpos up -d
+## 4. 주문 관리
+
+### 주문 생성
+- [ ] 주문 타입은 필수값이다.
+- [ ] 메뉴 주문은 필수값이며, 최소 1개 이상이어야 한다.
+- [ ] 메뉴 주문의 메뉴가 존재해야 한다.
+- [ ] 메뉴 주문 수량은 **포장 또는 배달 주문의 경우** 0보다 커야 한다.
+- [ ] 메뉴 주문의 메뉴는 **노출된 메뉴여야 한다**.
+- [ ] 메뉴 주문 가격은 메뉴 가격과 동일해야 한다.
+- [ ] 주문 생성 시 **주문 상태는 "대기"** 여야 한다.
+- [ ] 주문 타입이 **배달인 경우, 배달 주소는 필수값**이다.
+- [ ] 주문 타입이 **먹고가기인 경우, 주문 테이블이 존재해야 하며 비어 있어야 한다**.
+
```
주문에 관련된 요구사항 잘 작성해주셨습니다 👍 

다만, 주문 상태와 타입이 어떤것들이 있는지와 주문 타입별로 프로세스 흐름이 어떤지 한줄씩 작성해주시면 
이해하는데 더 도움이 될것 같아요 : )

예시) 주문 타입이 포장인 경우 주문 상태는 (대기 -> 접수 -> ... ) 순서로 변경된다

mkkim90 replied:
네넵 용어사전에 기입해두겠습니다

MaroKim commented on README.md:
```diff
@@ -8,7 +8,130 @@ docker compose -p kitchenpos up -d
+- [ ] 현재 주문 상태가 "대기"여야 한다.
+- [ ] 주문 타입이 **배달인 경우, 배달 접수 처리를 해야 한다**.
+
+### 주문 준비
+- [ ] 주문이 존재해야 한다.
+- [ ] 현재 주문 상태가 "접수"여야 한다.
+
```
주문 상태에 대해 잘 작성해주셨습니다 👍 

다만, 개인적으로 준비라는 용어 보다는 다른 단어를 한번 고민해주시면 어떨지 문의 드립니다.
코드상에서 serve라는 단어를 사용한 점과 해당 단어의 뜻을 고려해서 봐주시면 좋을 것 같아요 :)

mkkim90 replied:
음식 준비 상태를 의도한거긴한데.. 주문 제공? 보다는 서빙으로 할까 하는 데 어떠세요?

MaroKim commented on README.md:
```diff
@@ -8,7 +8,130 @@ docker compose -p kitchenpos up -d
+
+### 주문 테이블 착석
+- [ ] 주문 테이블이 존재해야 한다.
+- [ ] 자리 상태를 **꽉 참(사용 중)으로 변경**해야 한다.
```
주문 테이블의 상태를 '비움', '꽉참(사용중)'으로 정의해주신것 같아요 :)

다만 사용중을 의미하는 단어가 두가지를 사용해주셨는데 한가지로 통일을 해주셔야 혼동이 되지 않을것 같고 
꽉참이라는 단어는 여러가지로 해석이 될 수 있을것 같습니다( 사용중인가, 테이블이 만석인가) 
고려해서 한번 수정 부탁드립니다 :)

mkkim90 replied:
만석 좋네요!

MaroKim reviewed (changes requested):
안녕하세요 민균님 
전반적으로 작성 깔끔하게 잘 해주셨습니다. 👍 

약간만 보완해주시면 될것 같아요 :) 확인 부탁드리겠습니다.

mkkim90 committed:
feedback : 보완 사항 변경
- 상품 수정시 합산 가격 비교로 표현
- 메뉴 생성시, 합산 가격 비교 수정
- 주문 준비상태 -> 서빙으로 변경
- 테이블 꽉 참(사용중) 만석으로 용어 통일

mkkim90 commented:
성태님 꼼꼼하게 리뷰해주셔서 감사합니다 보완사항들 수정하여 반영하였습니다 🙏

MaroKim commented on README.md:
```diff
@@ -18,7 +18,7 @@ docker compose -p kitchenpos up -d
 - [ ] 수정하고자 하는 상품이 존재해야한다.
 - [ ] 수정한 가격이 해당 상품이 포함된 메뉴 가격에 반영되어야 한다. 
-  - [ ] 합산 가격이 일치하지 않으면 해당 상품을 비노출 처리한다.
+  - [ ] 메뉴 가격이 상품의 합산 가격보다 클 경우, 해당 상품을 비노출 처리한다.
```
메뉴 가격으로 인한  비노출의 기준이 명확해졌습니다 👍

MaroKim commented on README.md:
```diff
@@ -30,10 +30,10 @@ docker compose -p kitchenpos up -d
 - [ ] 메뉴 상품은 반드시 상품 테이블에 존재해야 한다.
 - [ ] 메뉴 상품의 수량은 0보다 커야한다.
-- [ ] 메뉴 가격은 메뉴 상품들의 가격 합과 일치해야 한다.
+- [ ] 메뉴 가격이 메뉴 상품들의 합산 가격보다 크지 않아야 한다. 
```
조건 수정 👍

MaroKim commented on README.md:
```diff
@@ -8,11 +8,136 @@ docker compose -p kitchenpos up -d
+### 메뉴 수정
+- [ ] 가격은 필수값이며, 0보다 커야 한다.
+- [ ] 수정하려는 메뉴가 존재해야 한다.
+- [ ] 수정된 가격이 메뉴 상품들의 가격 합과 일치해야 한다.
+
+### 메뉴 노출
+- [ ] 노출하려는 메뉴가 존재해야 한다.
+- [ ] 메뉴 가격이 메뉴 상품들의 가격 합과 일치해야 한다.
```
여기 가격 조건도 수정 부탁드립니다 :)

MaroKim commented on README.md:
```diff
@@ -8,11 +8,136 @@ docker compose -p kitchenpos up -d
+### 주문 배달
+- [ ] 주문이 존재해야 한다.
+- [ ] 주문 타입이 "배달"이어야 한다.
+- [ ] 현재 주문 상태가 "서빙"여야 한다.
```
서빙으로 변경해주셨네요 :)
먹고 가시는분에게도, 포장을 원하는 손님에게도, 그리고 배달라이더 분들에게 전달할때도 
사용하기에 적절한것 같습니다 😄

MaroKim commented on README.md:
```diff
@@ -8,11 +8,136 @@ docker compose -p kitchenpos up -d
+- [ ] 손님 수는 **0으로 초기 설정**한다.
+- [ ] 자리 상태는 **비움으로 초기 설정**한다.
+
+### 주문 테이블 착석
+- [ ] 주문 테이블이 존재해야 한다.
+- [ ] 자리 상태를 **만석으로 변경**해야 한다.
+
+### 주문 테이블 비우기
+- [ ] 주문 테이블이 존재해야 한다.
+- [ ] 손님 수를 **0으로 설정**해야 한다.
+- [ ] 자리 상태를 **비움으로 변경**해야 한다.
+- [ ] 주문 상태가 **완료된 상태**여야 한다.
+
+### 주문 테이블 손님 수 변경
+- [ ] 손님 수는 0보다 커야 한다.
+- [ ] 주문 테이블이 존재해야 한다.
+- [ ] 자리 상태가 **만석 상태**여야 한다.
+
```
제가 설명이 부족했던 것 같습니다. 🙇‍♂️ 

단어를 하나로 통일해서 수정해주신 점 👍 

기존에 '꽉 찬' 이라는 단어가 표현이 애매한 점을 말씀드렸습니다.
지금은 '만석'이라는 단어로 변경해주셨는데요
비움 상태가 아닐때 만석 상태인데 이미 주문테이블의 상태가 만석상태일때
손님수가 0이 아닌 수로 변경이 되어도 '만석'이라는 단어를 사용하는데요 

만석의 의미는 '가득찬' 의미가 더 커보입니다. 다른 단어를 한번 고민해주시면 감사하겠습니다 🙇‍♂️ 
간단하게 '사용중'이라는 단어도 괜찮을것 같습니다

MaroKim commented on README.md:
```diff
@@ -8,11 +8,136 @@ docker compose -p kitchenpos up -d
-| 한글명 | 영문명 | 설명 |
-| --- | --- | --- |
-|  |  |  |
+| **한글명**   | **영문명**  | **설명** |
+|------------|------------|----------------------------------------------------------------|
+| **포장**    | TAKEOUT    | 음식을 가져가는 주문 타입. 주문 상태: `대기 → 접수 → 서빙 → 최종완료` |
+| **먹고가기** | EAT_IN     | 매장에서 먹고 가는 주문 타입. 주문 상태: `대기 → 접수 → 서빙 → 최종완료` |
+| **배달**    | DELIVERY   | 고객 주소로 배달하는 주문 타입. 주문 상태: `대기 → 접수 → 서빙 → 배달중 → 배달완료 → 최종완료` | 
```
각 주문 타입별로 상태 값을 용어 사전에 작성해주셨네요 

저는 주문 소제목 아래에 설명을 작성해주시면 좋을것 같다라는 느낌으로 말씀드렸는데 
제가 혼란을 드렸네요 ㅠ 죄송합니다.

제가 말씀드린건 이해하기 위한 설명정도로써 사용하기 좋은거라 필수는 아닐것 같습니다. :)

용어사전은 다음 수업인 '크게 소리 내어 모델링 하기'의 과제로 작성을 하게 되시는데
미리 경험을 해보신건데 잘 작성해주셨어요 👍

MaroKim reviewed (approved):
민균님 빠르게 수정해주셨네요 👍 
몇가지 코멘트를 드렸는데 많지 않고 금방 수정하실수 있을 것 같아서 머지하겠습니다.

다음 과제 진행하시기 전에 확인만 부탁드리겠습니다 :)
고생하셨습니다.
도움이 필요하시거나 추가 설명이 필요하시면 언제든지 DM 부탁드립니다 😄
